gsa/src/web/pages/audits/__tests__/listpage.js

```diff
@@ -0,0 +1,187 @@
+  target: {_id: 'id1', name: 'target1'},
+});
+
+const caps = new Capabilities(['everything']);
+const wrongCaps = new Capabilities(['get_config']);
+
+const reloadInterval = 1;
+const manualUrl = 'test/';
+
+const currentSettings = jest.fn().mockReturnValue(
+  Promise.resolve({
+    foo: 'bar',
+  }),
+);
+
+const getFilters = jest.fn().mockReturnValue(
+  Promise.resolve({
+    data: [],
+    meta: {
+      filter: Filter.fromString(),
+      counts: new CollectionCounts(),
+    },
+  }),
+);
+
+const getAudits = jest.fn().mockReturnValue(
+  Promise.resolve({
+    data: [audit],
+    meta: {
+      filter: Filter.fromString(),
+      counts: new CollectionCounts(),
+    },
+  }),
+);
+
+const getReportFormats = jest.fn().mockReturnValue(
+  Promise.resolve({
+    data: [],
+    meta: {
+      filter: Filter.fromString(),
+      counts: new CollectionCounts(),
+    },
+  }),
+);
+
+describe('AuditPage tests', () => {
+  test('should render full AuditPage', () => {
+    const gmp = {
+      audits: {
+        get: getAudits,
+      },
+      filters: {
+        get: getFilters,
+      },
+      reportformats: {
+        get: getReportFormats,
+      },
+      reloadInterval,
+      settings: {manualUrl},
+      user: {currentSettings},
+    };
+
+    const {render, store} = rendererWith({
+      gmp,
+      capabilities: true,
+      store: true,
+      router: true,
+    });
+
+    store.dispatch(setUsername('admin'));
+
+    const counts = new CollectionCounts({
+      first: 1,
+      all: 1,
+      filtered: 1,
+      length: 1,
+      rows: 10,
+    });
+    const filter = Filter.fromString('first=1 rows=10');
+    const loadedFilter = Filter.fromString('first=1 rows=10');
+    store.dispatch(
+      entitiesActions.success([audit], filter, loadedFilter, counts),
+    );
+
+    const {baseElement} = render(<AuditPage />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+});
+
+describe('AuditPage ToolBarIcons test', () => {
+  test('should render', () => {
+    const handleAuditCreateClick = jest.fn();
+
+    const {render} = rendererWith({
+      capabilities: caps,
+      router: true,
+    });
+
+    const {element} = render(
+      <ToolBarIcons
+        entity={audit}
+        onAuditCreateClick={handleAuditCreateClick}
+      />,
+    );
+    expect(element).toMatchSnapshot();
+  });
+
+  test('should call click handlers', () => {
+    const handleAuditCreateClick = jest.fn();
+
+    const {render} = rendererWith({
+      capabilities: caps,
+      router: true,
+    });
+
+    const {getAllByTestId} = render(
+      <ToolBarIcons
+        entity={audit}
+        onAuditCreateClick={handleAuditCreateClick}
+      />,
+    );
+
+    const icons = getAllByTestId('svg-icon');
+
+    fireEvent.click(icons[0]);
+    expect(handleAuditCreateClick).toHaveBeenCalled();
+    expect(icons[0]).toHaveAttribute('title', 'New Audit');
+  });
+
+  test('should not call click handlers without permission', () => {
```

Comment:
This is not really tested here. You are testing that an icon is not rendered when the permissions/caps are wrong. depending on the icon's behavior either the test or the title should be adjusted (in this case it's the title, I guess).